### PR TITLE
Use custom VTK with OSMesa wheel to build the docs

### LINKED ISSUE #1854: documentation not building
**Describe the bug, what's wrong, and what you expect:**

I cannot get the documentation to build locally. The examples gallery builds fine and gets cached fine) but numpydoc seems to be blowing it up.

It helps dramatically to comment this line out:

https://github.com/pyvista/pyvista/blob/5dd060e465bd8339346de677d7b2b5aa661344af/doc/conf.py#L352

But it is still very slow during the "Reading sources..." steps.

Furthermore, when it does finish that, I am getting other errors which I'm currently tracking down

-----

**To Reproduce**

Please include a code snippet to reproduce the bug in the block below:

```py
cd doc
make html
```
-----

**System Information:**
 
```txt
Python 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:50:38)
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvista
>>> pyvista.Report()

--------------------------------------------------------------------------------
  Date: Thu Nov 25 22:45:22 2021 MST

                OS : Darwin
            CPU(s) : 12
           Machine : x86_64
      Architecture : 64bit
               RAM : 32.0 GiB
       Environment : Python
       File system : apfs
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon Pro Vega 20 OpenGL Engine
       GPU Version : 4.1 ATI-4.7.29

  Python 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:50:38)
  [Clang 11.1.0 ]

           pyvista : 0.32.dev0
               vtk : 9.1.0
             numpy : 1.21.4
           imageio : 2.12.0
           appdirs : 1.4.4
            scooby : 0.5.9
            meshio : 4.4.6
        matplotlib : 3.5.0
           IPython : 7.29.0
          colorcet : 1.0.0
           cmocean : 2.0
        ipyvtklink : 0.2.1
             scipy : 1.7.3
              tqdm : 4.62.3
--------------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by akaszynski**
Fixes documentation intermittent failures.

Avoid using `xvfb` when building our documentation by using VTK built with OSMesa. It's more stable, and as this PR shows, the documentation step was successful without intermittent failures (red X's are due to style guide failures).

Wheel built by following:
https://github.com/pyvista/pyvista-wheels

It also resolves #1854 by changing `extract-edges.py` to use a different mesh to avoid the memory error. This will have to be fixed in VTK.

### CODE REVIEW

**Comment by akaszynski** [comment]
Guess extract edges still fails. There’s some weird hardware issue going on.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -132,6 +128,11 @@ jobs:
       path: '$(Build.SourcesDirectory)/doc/examples'
     displayName: Build pyvista doc examples cache
 
+  # note: built this wheel using Ubuntu 20.04.3 with OSMESA
+  - script: |
+      pip install https://github.com/pyvista/pyvista-wheels/raw/add_v9_1_0/vtk-osmesa-9.1.0-cp38-cp38-linux_x86_64.whl
+    displayName: "Install VTK build with osmesa 9.1.0"
+
```
Comment (akaszynski) [inline azure-pipelines.yml:135]:
```suggestion
 
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -132,6 +128,11 @@ jobs:
...
```
Comment (akaszynski) [inline azure-pipelines.yml:135]:
```suggestion

```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -101,13 +101,9 @@ jobs:
...
-      .ci/setup_headless_display.sh
-      # python .ci/pyvista_test.py  # for debug
-    displayName: Install headless display
-
   - script: |
-      sudo apt-get install python3-tk pandoc
+      sudo apt-get update
+      sudo apt-get install python3-tk pandoc libosmesa6
       pip install -r requirements_docs.txt
     displayName: 'Install dependencies'
```
Comment (akaszynski) [inline azure-pipelines.yml:109]:
```suggestion
 
```

**Comment by adeak** [comment]
Sorry @akaszynski but I don't have the knowledge to review the CI changes which are the main point of this PR. Example changes seem OK at a glance but I haven't built it myself.

**Comment by akaszynski** [comment]
No worries @adeak. The build has worked 8 out of 8 times without retry, and I've verified the docs locally as well.

Considering how the doc-build is failing almost half the time, this is needed for us to get our CI/CD working smoothly and consistently.

**Pull request merged**